wavepool/admin.py

```diff
@@ -7,6 +7,21 @@ class NewsPostForm(forms.ModelForm):
     model = NewsPost
     fields = '__all__'
 
+    def clean(self, *args, **kwargs):
+        """ cleans form data
+        """
+        # user selected "is cover story checkbox":
+        if self.data.get('is_cover_story') == 'on':
+            all_newsposts = NewsPost.objects.all()
+            # for each newspost, if it is the current cover story and NOT this newspost anyways,
+            # unset it as cover story and save
+            for n in all_newsposts:
+                if n.is_cover_story == True:
+                    if n != self.instance:
+                        n.is_cover_story = False
+            n.save()
```

Comment:
This can be consolidated into a pretty straightforward django query. 
Assuming we are in a part of the code where we have the instance, we can do something similar to:
```
if instance.is_cover_story:
    NewsPost.objects.filter(is_cover_story=True).exclude(pk=instance.pk)
```